packages/core/src/components/axes/axis.ts

```diff
@@ -472,8 +511,7 @@ export class Axis extends Component {
 					: estimatedTickSize < minTickSize;
 			}
 
-			// always rotate ticks if zoomDomain is changing to avoid rotation flips during zoomDomain changing
-			if (rotateTicks || this.zoomDomainChanging) {
+			if (this.isTickRotationRequired(rotateTicks)) {
```

Comment:
why would we go through labels and calculate whether any need rotation before knowing whether we are rotating or not?

Reply:
You are right, if we already know that we need to rotate or not, we could avoid some calculation

Updated in https://github.com/carbon-design-system/carbon-charts/pull/770/commits/f2489e73a5a93494d165e1c23251f39f939b0338.